### DISCUSSION TITLE
DirectoryModuleCatalog.BuildChildDomain missing when upgrading to .NET 5

### DISCUSSION THREAD

**Discussion by obi111** [PrismLibrary/Prism#2272] [WPF]
I have upgraded our .net472 WPF project to .NET5 and now the DirectoryModuleCatalog.BuildChildDomain is not available any longer (still using Prism.WPF 7.2.0.1422)

The approach in described here: https://brianlagunas.com/prism-dynamically-discover-and-load-modules-at-runtime/

Does anyone know what is the issue here?

**Comment by obi111** [comment]
protected virtual AppDomain BuildChildDomain(AppDomain parentDomain)
      {
         if (parentDomain == null) throw new System.ArgumentNullException("parentDomain");

         Evidence evidence = new Evidence(parentDomain.Evidence);
         AppDomainSetup setup = parentDomain.SetupInformation;
         return AppDomain.CreateDomain("DiscoveryRegion", evidence, setup);
      }

AppDomain.Evidence is missing in .net 5 ...

**Reply by brianlagunas** [reply]
Correct. You'll have to modify your code to make it work.

**Reply by obi111** [reply]
Sure do you have any hints for that? I am new to .net 5 ...

**Reply by obi111** [reply]
I think the AssemblyLoadContext class should be used instead?
